specs/network_details_for_hardware_inventory.adoc

```diff
@@ -0,0 +1,150 @@
+= Specification for adding network details in node hardware inventory
+
+Add network details also as a part of node hardware inventory
+
+== Problem description
+
+Hardware inventory details are used to show exact status of the particular
+node. Currently network details are missing in node hardware inventory.
+we have to add network details like network_id, ipv4, ipv6, netmask, status,
+interface_id, sysfs_id, device_link, interface_type, model, driver_modules, drive
+hw_address, link_detected as a part of node hardware inventory.
+
+== Use Cases
+
+Tendrl should show the current status of network interface.
+
+== Proposed change
+* From node agent collect all network interface details in node inventory list and
+  store it in etcd. (Detailed information in implementation section).
+
+== Alternatives
+
+* We can also use ethtool for getting network details. But ethool is not
+  available for pip.
+
+== Data model impact
+
+* Add network details in etcd with key nodes/node_id/Networks/interface_name.
+
+== Impacted modules
+
+=== Tendrl/common impact
+
+None
+
+=== Tendrl/node_agent impact
+
+* Add a new function in pull_hardware_inventory to collect network interface
+  details.
+* Create a new class called Network in persistence module to keep the network
+  details.
+* Add a new function in persister to persist network details.
+* Modify the manager modules to read and store network interface details.
+* Create an object in tendrl_definitions_node_agent for network details.
+* Commands used to retrieve network details are:
+```
+    interfaces = netifaces.interfaces()
+    ipv4_addr_list = netifaces.ifaddresses(interfaces[integer_value])[AF_INET]
+    netmask = ipv4_addr_detail[integer_value]['netmask']
+    ipv6_addr_list = netifaces.ifaddresses(interfaces[integer_value])[AF_INET6]
+```
+* Commnad for check network interface is up or not
+```
+    Command({"_raw_params": "cat /sys/class/net/wlp3s0/operstate"})
+    Command({"_raw_params": 'hwinfo --network'})
+```
+* Expected output format for the network details is:
+```
+{"interface_name": {"ipv4": ["ipv4address", ...],
+                    "ipv6": ["ipv6address, ..."],
+                    "netmask": ["subnet", ...],
+                    "status":"up/down",
+                    "interface_id": "",
+                    "sysfs_id": "",
+                    "device_link": "",
+                    "interface_type": "",
+                    "model": "",
+                    "driver_modules": "",
+                    "drive": "",
+                    "hw_address": "",
+                    "link_detected": ""
+                   }, ...
+}
+```
+* Add new object in tendrl defenition file with list.
+```
+list: nodes/$Node_context.node.id/Network
```

Comment:
What is the exact structure of the data in etcd? I think that there needs to be a global `networks` hierarchy as well, where we maintain a list of networks, under which we point to each node that belongs to this network. This is in addition to the network details maintained under each node.